Fix docker postgres issues

benjie opened the pull request:
Fixes #110 


@JoeSchr can you cast your eyes over this? The `postgres:11` image now requires the `POSTGRES_PASSWORD` envvar to be set. I've made this work with the standard `yarn docker setup && yarn docker start` flow, but not sure if we need to do anything to make it work with devcontainers?

JoeSchr commented on docker/docker.env:
```diff
@@ -7,9 +7,6 @@ FORCE_COLOR=2
 
-# Instructs to use our docker PostgreSQL service.
-DATABASE_HOST=db
-ROOT_DATABASE_URL=postgres://postgres@db/template1
```
I don't think we should remove that, because it's there to override the .env vars when using docker-compose

JoeSchr commented on docker/scripts/yarn-setup.js:
```diff
@@ -1,6 +1,10 @@
+const { safeRandomString } = require("../../scripts/lib/random");
+const fsp = require("fs").promises;
+
+const DOTENV_PATH = `${__dirname}/../../.env`;
```
or we should just use `docker.env` here instead of `.env`

why does it need to be in `.env`? it's only relevant for docker right?

benjie replied:
Basically: `docker.env` is committed, `.env` is not, and we need `ROOT_DATABASE_URL` to contain the password we've generated for `POSTGRES_PASSWORD` so we cannot write it to `docker.env` otherwise it'd be tracked under git. So instead I just have Postgres source the `.env` file.

Oh... I see what you mean - this breaks dual Docker/local dev. I don't think that's an issue, TBH - we've told users to pick one and stick to it in the README so it should be fine? Maybe we should add a `docker/.env` file that contains these secrets and source both, though. It looks like (though the docs don't say) `env_file` could take an array. I'm going to give this a try.

benjie commented:
@JoeSchr based on your feedback, I've removed `docker/docker.env` and replaced it with `docker/.env` which is a secret file (untracked) that the `yarn docker setup` command populates automatically. I did consider keeping both a tracked and untracked .env but I decided that was too much.

JoeSchr commented:
I'm kind of using dual docker/local mode all the time nowadays, so I would
be glad if it didn't break. but if it's needed I will work around it


__

Typed on a mobile, please excuse any typing mistakes or short-spokeness.

On Wed, Feb 26, 2020, 16:16 Benjie Gillam <notifications@github.com> wrote:

> *@benjie* commented on this pull request.
> ------------------------------
>
> In docker/scripts/yarn-setup.js
> <https://github.com/graphile/starter/pull/117#discussion_r384558268>:
>
> > @@ -1,6 +1,10 @@
>  const { spawnSync: baseSpawnSync } = require("child_process");
>  const { basename, dirname, resolve } = require("path");
>  const platform = require("os").platform();
> +const { safeRandomString } = require("../../scripts/lib/random");
> +const fsp = require("fs").promises;
> +
> +const DOTENV_PATH = `${__dirname}/../../.env`;
>
> Basically: docker.env is committed, .env is not, and we need
> ROOT_DATABASE_URL to contain the password we've generated for
> POSTGRES_PASSWORD so we cannot write it to docker.env otherwise it'd be
> tracked under git. So instead I just have Postgres source the .env file.
>
> Oh... I see what you mean - this breaks dual Docker/local dev. I don't
> think that's an issue, TBH - we've told users to pick one and stick to it
> in the README so it should be fine? Maybe we should add a docker/.env
> file that contains these secrets and source both, though. It looks like
> (though the docs don't say) env_file could take an array. I'm going to
> give this a try.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/graphile/starter/pull/117?email_source=notifications&email_token=AB6WSHWIIZ7G4J2TCLYJECTRE2BTLA5CNFSM4K4C3PW2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCXATMJY#discussion_r384558268>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6WSHSZ5LFTGKTN7VUEQRDRE2BTLANCNFSM4K4C3PWQ>
> .
>

benjie commented:
I think the above solves this :+1: Let me know if it doesn't.

JoeSchr commented:
I think so too, just wanted to stated it as an use case that's actively used

On Thu, Feb 27, 2020 at 1:04 PM Benjie Gillam <notifications@github.com>
wrote:

> I think the above solves this 👍 Let me know if it doesn't.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/graphile/starter/pull/117?email_source=notifications&email_token=AB6WSHU5EMRFHMWEGL2XYU3RE6T4BA5CNFSM4K4C3PW2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENEDXNI#issuecomment-591936437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6WSHVOOFHVFKIKNXAN4JTRE6T4BANCNFSM4K4C3PWQ>
> .
>
